src/Featurizers/ForecastingPivotFeaturizer.h

```diff
@@ -66,27 +68,29 @@ class ForecastingPivotTransformer :
     // MSVC runs into problems when the declaration and definition are separated
     void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
         if (input.empty()) {
-            return;
+            throw std::invalid_argument("There's no input matrix passed in!");
         }
         
-        std::vector<typename Microsoft::Featurizer::Traits<InputT>::nullable_type> ret;
         Eigen::Index const col_size(input[0].cols());
 
         for (Eigen::Index col = 0; col < col_size; ++col) {
+            std::vector<InputT> ret;
             bool has_nan(false);
 
             // if any nans is found in the current column across all matrixes, this column is omitted
             for (MatrixType const & matrix : input) {
                 // all matrixes must have the same column number
-                assert(matrix.cols() == col_size);
+                if (matrix.cols() != col_size) {
+                    throw std::invalid_argument("All input matrixes should have the same number of columns!");
+                }
 
                 for (Eigen::Index row = 0; row < matrix.rows(); ++row) {
                     if (Traits<NullableType>::IsNull(matrix(row, col))) {
                         has_nan = true;
                         break;
                     }
 
-                    ret.emplace_back(matrix(row, col));
+                    ret.emplace_back(Traits<NullableType>::GetNullableValue(matrix(row, col)));
```

Comment:
Do you need to use `Traits<NullableType>::GetNullableValue` here? The vector is defined of type `InputT`, and if there is a null value we aren't outputting the row.

Reply:
I think `GetNullableValue` is necessary here because `nullable_type` can be nonstd::optional<T> or double_t and their `GetNullableValue` are implemented differently. In addition, I don't think it's convertible from a, say nonstd::optional<string> to string directly.